libibverbs/cmd.c

```diff
@@ -265,6 +265,15 @@ int ibv_cmd_query_device_ex(struct ibv_context *context,
 		}
 	}
 
+	if (attr_size >= offsetof(struct ibv_device_attr_ex, xrc_odp_caps) +
```

Comment:
All these attr_size checks are obsolete and should be completely deleted. Now that verbs and the providers share the same code it is not possible for attr to not have xrc_odp_caps.

Reply:
'attr_size' is the application's input size which comes from the static inline function
ibv_query_device_ex(),  applications that were complied with legacy libibverbs may not have this field in place, that's why this check is still needed here as done also above.